edit_course/views.py

```diff
@@ -122,6 +124,22 @@ def renumber_recursion(self, parent, n=1):
             n += 1
         return n
 
+    def number_learning_objects_consecutively(self) -> None:
+        """Number all learning objects (i.e. chapters and exercises)"""
+        def inner(parent: Union[CourseModule, CourseChapter], n: int):
+            for entry in parent['children']:
+                model = LearningObject.objects.get(id=entry['id'])
```

Comment:
Model objects (CourseModule, CourseChapter) should be indexed using `.attribute` instead of `['attribute']`. The objects.get is also unnecessary after the function actually takes model objects because iterating `parent.children.all()` will give you actual objects.

Reply:
If the order is calculated dynamically, you can just use the cache directly, so there are no need to use model objects here.